src/transformers/configuration_fsmt.py

```diff
@@ -0,0 +1,211 @@
+        encoder_attention_heads (:obj:`int`, optional, defaults to 16):
+            Number of attention heads for each attention layer in the Transformer encoder.
+        decoder_attention_heads (:obj:`int`, optional, defaults to 16):
+            Number of attention heads for each attention layer in the Transformer decoder.
+        decoder_ffn_dim (:obj:`int`, optional, defaults to 4096):
+            Dimensionality of the "intermediate" (i.e., feed-forward) layer in decoder.
+        encoder_ffn_dim (:obj:`int`, optional, defaults to 4096):
+            Dimensionality of the "intermediate" (i.e., feed-forward) layer in decoder.
+        activation_function (:obj:`str` or :obj:`function`, optional, defaults to "relu"):
+            The non-linear activation function (function or string) in the encoder and pooler.
+            If string, "gelu", "relu", "swish" and "gelu_new" are supported.
+        dropout (:obj:`float`, optional, defaults to 0.1):
+            The dropout probabilitiy for all fully connected layers in the embeddings, encoder, and pooler.
+        attention_dropout (:obj:`float`, optional, defaults to 0.0):
+            The dropout ratio for the attention probabilities.
+        activation_dropout (:obj:`float`, optional, defaults to 0.0):
+            The dropout ratio for activations inside the fully connected layer.
+        max_position_embeddings (:obj:`int`, optional, defaults to 1024):
+            The maximum sequence length that this model might ever be used with.
+            Typically set this to something large just in case (e.g., 512 or 1024 or 2048).
+        init_std (:obj:`float`, optional, defaults to 0.02):
+            The standard deviation of the truncated_normal_initializer for initializing all weight matrices.
+        scale_embedding (:obj:`bool`, optional, defaults to :obj:`True`):
+            Scale embeddings by diving by sqrt(d_model).
+        bos_token_id (:obj:`int`, optional, defaults to 0)
+            Beginning of stream token id.
+        pad_token_id (:obj:`int`, optional, defaults to 1)
+            Padding token id.
+        eos_token_id (:obj:`int`, optional, defaults to 2)
+            End of stream token id.
+        decoder_start_token_id (:obj:`int`, `optional`):
+            This model starts decoding with `eos_token_id`
+        encoder_layerdrop: (:obj:`float`, optional, defaults to 0.0):
+            Google "layerdrop arxiv", as its not explainable in one line.
+        decoder_layerdrop: (:obj:`float`, optional, defaults to 0.0):
+            Google "layerdrop arxiv", as its not explainable in one line.
+        is_encoder_decoder (:obj:`bool`, optional, defaults to :obj:`True`):
+            Whether this is an encoder/decoder model.
+        tie_word_embeddings (:obj:`bool`, optional, defaults to :obj:`False`):
+            Whether to tie input and output embeddings.
+"""
+
+
+class DecoderConfig(PretrainedConfig):
```

Comment:
can we just have `config.decoder_vocab_size`, `config.decoder_bos_token_id`?

Reply:
Hmm, I didn't see that comment until now. We sure can, but that would be yet another way - won't we want things to be consistent. I suppose we need more similar models to find how we want it. 

I will leave it for now and then we can tweak later - it won't impact the user.